packages/cli/src/commands/runAndroid/__tests__/getLaunchPackageName.test.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+import fs from 'fs';
+import getLaunchPackageName from '../getLaunchPackageName';
+
+jest.mock('fs');
+jest.mock('path');
+
+describe('run-android::getLaunchPackageName', () => {
+  const actualFs = jest.requireActual('fs');
+  const actualPath = jest.requireActual('path');
+
+  fs.readFileSync = jest.fn(filename => {
+    switch (filename) {
+      case 'app/build.gradle':
```

Comment:
this is throwing for me when I run the tests
```
(node:8120) UnhandledPromiseRejectionWarning: Error: ENOENT: no such file or directory, open 'app/build.gradle'
    at Object.openSync (fs.js:439:3)
    at Object.readFileSync (fs.js:344:35)
    at findBuildTypes (/Users/bartolkaruza/dev/proj/oss/react-native-cli/packages/cli/src/commands/runAndroid/getLaunchPackageName.js:58:33)
    at splitVariant (/Users/bartolkaruza/dev/proj/oss/react-native-cli/packages/cli/src/commands/runAndroid/getLaunchPackageName.js:96:22)
    at getManifestFile (/Users/bartolkaruza/dev/proj/oss/react-native-cli/packages/cli/src/commands/runAndroid/getLaunchPackageName.js:136:7)
    at getLaunchPackageName (/Users/bartolkaruza/dev/proj/oss/react-native-cli/packages/cli/src/commands/runAndroid/getLaunchPackageName.js:155:24)
    at runOnAllDevices (/Users/bartolkaruza/dev/proj/oss/react-native-cli/packages/cli/src/commands/runAndroid/runOnAllDevices.js:83:99)
    at Object.it (/Users/bartolkaruza/dev/proj/oss/react-native-cli/packages/cli/src/commands/runAndroid/__tests__/runOnAllDevices.test.js:29:34)
    at Object.asyncJestTest (/Users/bartolkaruza/dev/proj/oss/react-native-cli/node_modules/jest-jasmine2/build/jasmineAsyncInstall.js:100:37)
    at resolve (/Users/bartolkaruza/dev/proj/oss/react-native-cli/node_modules/jest-jasmine2/build/queueRunner.js:41:12)
```

Reply:
The failing promises are on other tests than the current one, so try running all tests for cli to reproduce. (yarn test) Output should be clean success for all tests, as it is on current master.

Reply:
I've pushed a commit that updates the tests that indirectly call getLaunchPackage to also mock the fs there, as that was causing the issue. However, it appears there's a Windows specific issue for the getDependencyConfig test: It's comparing a *nix path in the snapshot to a Windows generated one, which will always fail.

Here is the output under Windows/Powershell
```
PS ...> yarn test
yarn run v1.13.0
$ jest
 PASS  packages/cli/src/tools/__tests__/findPlugins-test.js
 FAIL  packages/cli/src/commands/link/__tests__/getDependencyConfig-test.js
  ● getDependencyConfig › should return an array of dependencies' config

    expect(value).toMatchSnapshot()

    Received value does not match stored snapshot "getDependencyConfig should return an array of dependencies' config 1".

    - Snapshot
    + Received

    @@ -9,7 +9,7 @@
            "sampleiOSKey": "",
          },
        },
        "name": "react-native-windows",
        "params": Array [],
    -   "path": "/root/node_modules/react-native-windows",
    +   "path": "\\root\\node_modules\\react-native-windows",
      }

      35 |     );
      36 | 
    > 37 |     expect(dependencies).toMatchSnapshot();
         |                          ^
      38 |   });
      39 | 
      40 |   it('should throw on invalid react-native dependency', () => {

      at Object.toMatchSnapshot (src/commands/link/__tests__/getDependencyConfig-test.js:37:26)

  ● getDependencyConfig › should throw on invalid react-native dependency

    expect(received).toThrowError()

    Received function did not throw

      41 |     expect(() =>
      42 |       getDependencyConfig({root: '/root'}, platforms, 'abcd'),
    > 43 |     ).toThrowError();
         |       ^
      44 |   });
      45 | });
      46 | 

      at Object.toThrowError (src/commands/link/__tests__/getDependencyConfig-test.js:43:7)

 › 1 snapshot failed.
 PASS  packages/cli/src/commands/upgrade/__tests__/upgrade.test.js
 PASS  packages/cli/src/commands/runAndroid/__tests__/runOnAllDevices.test.js
 PASS  packages/cli/src/tools/__tests__/android/getProjectConfig-test.js
 PASS  packages/cli/src/tools/__tests__/ios/getProjectConfig-test.js
 PASS  packages/cli/src/commands/link/__tests__/link-test.js
 PASS  packages/cli/src/commands/link/__tests__/ios/createGroup-test.js
 PASS  packages/cli/src/commands/link/__tests__/ios/removeSharedLibrary-test.js
 PASS  packages/cli/src/commands/info/__tests__/info.test.js
 PASS  packages/cli/src/tools/__tests__/ios/findProject-test.js
 PASS  packages/cli/src/commands/link/__tests__/ios/getPlist-test.js
 PASS  packages/cli/src/commands/link/__tests__/ios/addSharedLibraries-test.js
 PASS  packages/cli/src/commands/link/__tests__/ios/writePlist-test.js
 PASS  packages/cli/src/commands/link/__tests__/ios/mapHeaderSearchPaths-test.js
 PASS  packages/cli/src/commands/link/__tests__/ios/addProjectToLibraries-test.js
 PASS  packages/cli/src/commands/link/__tests__/ios/getPlistPath-test.js
 PASS  packages/cli/src/commands/link/__tests__/ios/removeProjectFromProject-test.js
 PASS  packages/cli/src/commands/link/__tests__/pods/findPodTargetLine-test.js
 PASS  packages/cli/src/commands/link/__tests__/ios/getGroup-test.js
 PASS  packages/cli/src/commands/link/__tests__/ios/isInstalled-test.js
 PASS  packages/cli/src/tools/__tests__/android/getDependencyConfig-test.js
 PASS  packages/cli/src/commands/link/__tests__/ios/getBuildProperty-test.js
 PASS  packages/cli/src/commands/server/debugger-ui/__tests__/DeltaPatcher-test.js
 PASS  packages/cli/src/tools/__tests__/findSymlinkedModules-test.js
 PASS  packages/cli/src/commands/link/__tests__/ios/hasLibraryImported-test.js
 PASS  packages/cli/src/tools/__tests__/android/readManifest-test.js
 PASS  packages/cli/src/tools/__tests__/android/findAndroidAppFolder-test.js
 PASS  packages/cli/src/commands/link/__tests__/ios/addFileToProject-test.js
 PASS  packages/cli/src/commands/link/__tests__/android/normalizeProjectName-test.js
 PASS  packages/cli/src/commands/link/__tests__/ios/getTargets-test.js
 PASS  packages/cli/src/commands/link/__tests__/ios/removeProjectFromLibraries-test.js
 PASS  packages/cli/src/tools/__tests__/ios/findPodspecName-test.js
 PASS  packages/cli/src/commands/link/__tests__/pods/isInstalled-test.js
 PASS  packages/cli/src/tools/__tests__/PackageManager-test.js
 PASS  packages/cli/src/commands/runAndroid/__tests__/getLaunchPackageName.test.js
 PASS  packages/cli/src/tools/__tests__/makeCommand-test.js
 PASS  packages/cli/src/tools/__tests__/findAssets-test.js
 PASS  packages/cli/src/commands/link/__tests__/android/makeStringsPatch-test.js
 PASS  packages/cli/src/tools/__tests__/android/findPackageClassName-test.js
 PASS  packages/cli/src/commands/link/__tests__/groupFilesByType-test.js
 PASS  packages/cli/src/tools/__tests__/android/findManifest-test.js
 PASS  packages/cli/src/commands/link/__tests__/ios/getHeaderSearchPath-test.js
 PASS  packages/cli/src/commands/link/__tests__/android/applyPatch-test.js
 PASS  packages/cli/src/commands/runIOS/__tests__/findMatchingSimulator-test.js
 PASS  packages/cli/src/commands/link/__tests__/promiseWaterfall-test.js
 PASS  packages/cli/src/commands/link/__tests__/android/makeSettingsPatch-test.js
 PASS  packages/cli/src/commands/bundle/__tests__/getAssetDestPathAndroid-test.js
 PASS  packages/cli/src/commands/link/__tests__/android/isInstalled-test.js
 PASS  packages/cli/src/commands/link/__tests__/ios/getHeadersInFolder-test.js
 PASS  packages/cli/src/commands/link/__tests__/android/makeBuildPatch-test.js
 PASS  packages/cli/src/tools/__tests__/ios/findPodfilePath-test.js
 PASS  packages/cli/src/commands/link/__tests__/android/makePackagePatch-test.js
 PASS  packages/cli/src/commands/bundle/__tests__/getAssetDestPathIOS-test.js
 PASS  packages/cli/src/commands/runIOS/__tests__/findXcodeProject-test.js
 PASS  packages/cli/src/commands/link/__tests__/pods/findMarkedLinesInPodfile-test.js
 PASS  packages/cli/src/commands/bundle/__tests__/filterPlatformAssetScales-test.js
 PASS  packages/cli/src/commands/link/__tests__/getProjectDependencies-test.js
 PASS  packages/cli/src/commands/link/__tests__/android/makeImportPatch-test.js
 PASS  packages/cli/src/commands/link/__tests__/pods/findLineToAddPod-test.js
 PASS  packages/cli/src/commands/runIOS/__tests__/parseIOSDevicesList-test.js
 PASS  packages/cli/src/commands/link/__tests__/pods/removePodEntry-test.js
 PASS  e2e/__tests__/install.test.js (7.307s)
 PASS  e2e/__tests__/uninstall.test.js (8.975s)

Summary of all failing tests
 FAIL  packages/cli/src/commands/link/__tests__/getDependencyConfig-test.js
  ● getDependencyConfig › should return an array of dependencies' config

    expect(value).toMatchSnapshot()

Snapshot Summary
 › 1 snapshot failed from 1 test suite. Inspect your code changes or run `yarn test -u` to update them.

Test Suites: 1 failed, 63 passed, 64 total
Tests:       2 failed, 2 todo, 225 passed, 229 total
Snapshots:   1 failed, 7 passed, 8 total
Time:        10.106s
Ran all test suites in 2 projects.
```

And the same test, ran under WSL/Ubuntu:
```
AnimaSA@DESKTOP-D0NNEGG /m/c/U/C/D/P/react-native-cli> yarn test
yarn run v1.13.0
$ jest
 PASS  packages/cli/src/commands/server/debugger-ui/__tests__/DeltaPatcher-test.js
 PASS  packages/cli/src/tools/__tests__/findPlugins-test.js
 PASS  packages/cli/src/tools/__tests__/findSymlinkedModules-test.js
 PASS  packages/cli/src/commands/runIOS/__tests__/findMatchingSimulator-test.js
 PASS  packages/cli/src/tools/__tests__/PackageManager-test.js
 PASS  packages/cli/src/commands/link/__tests__/android/makeSettingsPatch-test.js
 PASS  packages/cli/src/commands/link/__tests__/ios/getHeaderSearchPath-test.js
 PASS  packages/cli/src/commands/runAndroid/__tests__/runOnAllDevices.test.js
 PASS  packages/cli/src/tools/__tests__/ios/findProject-test.js
 PASS  packages/cli/src/commands/link/__tests__/android/makeBuildPatch-test.js
 PASS  packages/cli/src/commands/upgrade/__tests__/upgrade.test.js
 PASS  packages/cli/src/commands/bundle/__tests__/getAssetDestPathAndroid-test.js
 PASS  packages/cli/src/commands/link/__tests__/pods/removePodEntry-test.js
 PASS  packages/cli/src/commands/link/__tests__/pods/findLineToAddPod-test.js
 PASS  packages/cli/src/commands/link/__tests__/ios/writePlist-test.js
 PASS  packages/cli/src/tools/__tests__/android/getProjectConfig-test.js
 PASS  packages/cli/src/tools/__tests__/android/getDependencyConfig-test.js
 PASS  packages/cli/src/commands/link/__tests__/ios/createGroup-test.js
 PASS  packages/cli/src/commands/info/__tests__/info.test.js
 PASS  e2e/__tests__/uninstall.test.js (7.324s)
 PASS  e2e/__tests__/install.test.js (8.516s)

Test Suites: 64 passed, 64 total
Tests:       2 todo, 227 passed, 229 total
Snapshots:   8 passed, 8 total
Time:        16.193s
Ran all test suites in 2 projects.
Done in 17.94s.
chris@DESKTOP-D0NNEGG /m/c/U/C/D/P/react-native-cli> yarn test
yarn run v1.13.0
$ jest
 PASS  packages/cli/src/tools/__tests__/PackageManager-test.js
 PASS  packages/cli/src/tools/__tests__/findPlugins-test.js
 PASS  packages/cli/src/tools/__tests__/findSymlinkedModules-test.js
 PASS  packages/cli/src/commands/server/debugger-ui/__tests__/DeltaPatcher-test.js
 PASS  packages/cli/src/commands/runIOS/__tests__/findMatchingSimulator-test.js
 PASS  packages/cli/src/tools/__tests__/android/getProjectConfig-test.js
 PASS  packages/cli/src/commands/runAndroid/__tests__/runOnAllDevices.test.js
 PASS  packages/cli/src/commands/link/__tests__/ios/addSharedLibraries-test.js
 PASS  packages/cli/src/commands/link/__tests__/ios/writePlist-test.js
 PASS  packages/cli/src/commands/link/__tests__/ios/isInstalled-test.js
 PASS  packages/cli/src/tools/__tests__/ios/findProject-test.js
 PASS  packages/cli/src/commands/link/__tests__/ios/createGroup-test.js
 PASS  packages/cli/src/commands/link/__tests__/link-test.js
 PASS  packages/cli/src/commands/upgrade/__tests__/upgrade.test.js
 PASS  packages/cli/src/commands/link/__tests__/android/makeSettingsPatch-test.js
 PASS  packages/cli/src/commands/link/__tests__/ios/hasLibraryImported-test.js
 PASS  packages/cli/src/commands/link/__tests__/ios/removeProjectFromProject-test.js
 PASS  packages/cli/src/tools/__tests__/android/getDependencyConfig-test.js
 PASS  packages/cli/src/tools/__tests__/ios/getProjectConfig-test.js
 PASS  packages/cli/src/tools/__tests__/ios/findPodfilePath-test.js
 PASS  packages/cli/src/commands/bundle/__tests__/getAssetDestPathAndroid-test.js
 PASS  packages/cli/src/commands/link/__tests__/ios/mapHeaderSearchPaths-test.js
 PASS  packages/cli/src/commands/link/__tests__/getDependencyConfig-test.js
 PASS  packages/cli/src/commands/link/__tests__/ios/addFileToProject-test.js
 PASS  packages/cli/src/commands/link/__tests__/ios/removeProjectFromLibraries-test.js
 PASS  packages/cli/src/commands/link/__tests__/ios/getGroup-test.js
 PASS  packages/cli/src/commands/link/__tests__/ios/removeSharedLibrary-test.js
 PASS  packages/cli/src/tools/__tests__/android/findPackageClassName-test.js
 PASS  packages/cli/src/commands/link/__tests__/ios/addProjectToLibraries-test.js
 PASS  packages/cli/src/commands/link/__tests__/ios/getHeaderSearchPath-test.js
 PASS  packages/cli/src/commands/link/__tests__/ios/getTargets-test.js
 PASS  packages/cli/src/commands/link/__tests__/android/makeBuildPatch-test.js
 PASS  packages/cli/src/tools/__tests__/ios/findPodspecName-test.js
 PASS  packages/cli/src/commands/link/__tests__/ios/getPlist-test.js
 PASS  packages/cli/src/commands/link/__tests__/ios/getPlistPath-test.js
 PASS  packages/cli/src/tools/__tests__/findAssets-test.js
 PASS  packages/cli/src/tools/__tests__/android/readManifest-test.js
 PASS  packages/cli/src/commands/link/__tests__/pods/findLineToAddPod-test.js
 PASS  packages/cli/src/commands/link/__tests__/ios/getBuildProperty-test.js
 PASS  packages/cli/src/commands/link/__tests__/android/makePackagePatch-test.js
 PASS  packages/cli/src/commands/link/__tests__/pods/removePodEntry-test.js
 PASS  packages/cli/src/commands/link/__tests__/groupFilesByType-test.js
 PASS  packages/cli/src/commands/bundle/__tests__/filterPlatformAssetScales-test.js
 PASS  packages/cli/src/commands/runIOS/__tests__/parseIOSDevicesList-test.js
 PASS  packages/cli/src/commands/bundle/__tests__/getAssetDestPathIOS-test.js
 PASS  packages/cli/src/commands/runIOS/__tests__/findXcodeProject-test.js
 PASS  packages/cli/src/commands/link/__tests__/pods/isInstalled-test.js
 PASS  packages/cli/src/commands/link/__tests__/promiseWaterfall-test.js
 PASS  packages/cli/src/commands/link/__tests__/pods/findPodTargetLine-test.js
 PASS  packages/cli/src/tools/__tests__/android/findAndroidAppFolder-test.js
 PASS  packages/cli/src/commands/link/__tests__/android/isInstalled-test.js
 PASS  packages/cli/src/commands/link/__tests__/android/normalizeProjectName-test.js
 PASS  packages/cli/src/tools/__tests__/android/findManifest-test.js
 PASS  packages/cli/src/commands/link/__tests__/ios/getHeadersInFolder-test.js
 PASS  packages/cli/src/commands/link/__tests__/android/makeImportPatch-test.js
 PASS  packages/cli/src/commands/link/__tests__/pods/findMarkedLinesInPodfile-test.js
 PASS  packages/cli/src/commands/link/__tests__/getProjectDependencies-test.js
 PASS  packages/cli/src/commands/link/__tests__/android/makeStringsPatch-test.js
 PASS  packages/cli/src/tools/__tests__/makeCommand-test.js
 PASS  packages/cli/src/commands/link/__tests__/android/applyPatch-test.js
 PASS  packages/cli/src/commands/runAndroid/__tests__/getLaunchPackageName.test.js

Test Suites: 61 passed, 61 total
Tests:       1 todo, 220 passed, 221 total
Snapshots:   8 passed, 8 total
Time:        7.322s
Ran all test suites.
Done in 8.67s.
```